core/src/com/unciv/ui/screens/diplomacyscreen/DiplomacyScreen.kt

```diff
@@ -872,8 +872,37 @@ class DiplomacyScreen(
             declareWarButton.disable()
             declareWarButton.setText(declareWarButton.text.toString() + " ($turnsToPeaceTreaty${Fonts.turn})")
         }
-        declareWarButton.onClick {
-            ConfirmPopup(this, "Declare war on [${otherCiv.civName}]?", "Declare war") {
+        declareWarButton.onClick {            
+            // Tell the player who all will join the other side from defensive pacts
+            var defensivePactJoinText = ""
+            val otherCivDefensivePactList = otherCiv.diplomacy.values.filter { it.otherCiv() != viewingCiv
+                && it.diplomaticStatus == DiplomaticStatus.DefensivePact && !it.otherCiv().isAtWarWith(viewingCiv) }
+                .map { it.otherCiv() }.toMutableList()
+            // Go through and find all of the defensive pact chains and add them to the list
+            for (i in 0 until  otherCivDefensivePactList.count())
+                otherCivDefensivePactList.addAll(otherCivDefensivePactList[i].diplomacy.values
+                    .filter { !otherCivDefensivePactList.contains(it.otherCiv()) 
+                        && it.otherCiv() != viewingCiv && it.otherCiv() != otherCiv
+                        && !it.otherCiv().isAtWarWith(viewingCiv) }
+                    .map { it.otherCiv() })
+            for (thirdDefensivePactCiv in otherCivDefensivePactList) {
+                defensivePactJoinText += if (viewingCiv.knows(thirdDefensivePactCiv))
+                    "\n [${thirdDefensivePactCiv.civName}] will also join them in the war"
```

Comment:
Reminder that these are not yet wrapped in curly brackets, this will not translate

Reply:
I'm still working on it. Do you have any idea on how I can do the loop that increases in size as it is looped through? Everything seems to either throw an error or shortcut the loop to not include newly added elements. A standard for (int i = 0; i < othercivDefensivePactList.count; i++) would work perfectly, but is this acceptable?